frontend/src/routes/[[lang=locale]]/candidate/(protected)/questions/+page.svelte

```diff
@@ -1,38 +1,217 @@
+                  {question.info}
+                </Expander>
+
+                <div class="pt-10">
+                  <!-- This gives empty form label error from Wave Extension for every empty dot, but fix should come from LikertResponseButton -->
+                  <LikertResponseButtons
+                    name={question.id}
+                    mode="display"
+                    options={question.values}
+                    selectedKey={answers[question.id].key} />
+
+                  {#if translate(answers[question.id].openAnswer) !== ''}
+                    <div class="pt-10">
+                      <QuestionOpenAnswer
+                        >{translate(answers[question.id].openAnswer)}</QuestionOpenAnswer>
+                    </div>
+                  {/if}
+
+                  <div class="flex justify-center py-20">
+                    <Button
+                      text={getAnsweredButtonText().text}
+                      href={$getRoute({route: Route.CandAppQuestions, id: question.id})}
+                      icon={getAnsweredButtonText().icon}
+                      iconPos="left"></Button>
+                  </div>
+                </div>
+                {#if categoryQuestions[categoryQuestions.length - 1] !== question}
+                  <hr class="mt-40" />
+                {:else}
+                  <div class="mb-40" />
+                {/if}
+              </div>
+
+              <!-- Question not yet answered -->
+            {:else}
+              <div class="pt-40">
+                <div class="text-accent">
+                  {question.category}
+                </div>
+
+                <Expander title={question.text ?? ''} variant="question" titleClass="text-warning">
+                  {question.info}
+                </Expander>
+
+                <!-- Navigate to unsanswered question -->
+                {#if dataEditable}
+                  <a
+                    class="flex justify-center py-20"
+                    href={$getRoute({route: Route.CandAppQuestions, id: question.id})}>
+                    <Button
+                      text={$t('candidateApp.questions.answerButton')}
+                      class="w-full max-w-md bg-base-300" />
+                  </a>
+                {:else}
+                  <p class="p-10">{$t('candidateApp.questions.notAnswered')}</p>
+                {/if}
+                {#if categoryQuestions[categoryQuestions.length - 1] !== question}
+                  <hr class="mt-40" />
+                {:else}
+                  <div class="mb-40" />
+                {/if}
+              </div>
+            {/if}
+          {/each}
+        </Expander>
+      </div>
+    {/each}
+
+    <div class="flex w-full justify-center py-40">
+      <Button
+        text={$t('candidateApp.questions.return')}
+        variant="main"
+        href={$getRoute({route: Route.CandAppHome})} />
+    </div>
+  </BasicPage>
+{/if}
```

Comment:
Could this be divided into smaller individual components? I'd argue there is too much going on in this file. For example, the logic for answered questions and unanswered questions cases could be in its own component instead of being nested inside of two for loops here.